File: java/sui-entity/src/main/java/ru/smsoft/sui/suientity/config/Autoconfiguration.java
package ru.smsoft.sui.suientity.config;

import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;

@ComponentScan(basePackages = "ru.smsoft.sui.suientity")
@EntityScan("ru.smsoft.sui.suientity.entity")
@EnableJpaRepositories("ru.smsoft.sui.suientity.repository")
@EnableJpaAuditing
@EnableCaching
@Configuration
public class Autoconfiguration {}

File: java/sui-utils/sui-utils-kotlin/src/main/java/ru/sui/utils/kotlin/config/AutoConfiguration.kt
package ru.sui.utils.kotlin.config

import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration


@Configuration
@ComponentScan("ru.sui.utils.kotlin")
// Костыль, чтобы резолвить контекст в sui проектах
// Находится тут, т.к. sui-utils-kotlin есть во всех других sui-проектах
// Не влияет на работоспособность sui, т.к. EnableAutoConfiguration является частью SpringBootApplication
@EnableAutoConfiguration
class AutoConfiguration
